lib/transformers/form.js

```diff
@@ -50,6 +50,17 @@ function generateErrorSummary(transformationOrder, transformations) {
 function form({schemaKey, schema, options, transform, data, schemaErrors, fullUiSchema} = {}) {
     let opts = {};
     let pageHeading = schema.title;
+    let required = [];
+    const requiredKeys = function(obj) {
+        Object.keys(obj).forEach(key => {
+            if (key === 'required' && Array.isArray(obj[key])) {
+                required = [...new Set([...required, ...obj[key]])];
```

Comment:
for a performance boost, you could just `.push()` all the array items to the `required` variable. then later on create the `Set` from it. Avoids multiple operations to get the distinct values. in to an array.

````
required = [...required, ...obj[key];
...
return new Set([...required]);
````

Reply:
Changing this to a combination of `.push()` and `.flat()` as discussed, appeared to decrease the performance according to jsbench. Although again, you're talking milliseconds, nothing that would be perceivable to a human! I'm leaning towards leaving it as is!